components/tikv_util/src/metrics/process_linux.rs

```diff
@@ -21,45 +21,51 @@ pub fn monitor_process() -> Result<()> {
 pub struct ProcessCollector {
     pid: libc::pid_t,
     descs: Vec<Desc>,
-    cpu_total: Mutex<Counter>,
-    vsize: Gauge,
-    rss: Gauge,
-    start_time: Gauge,
+    cpu_total: Mutex<IntCounter>,
+    vsize: IntGauge,
+    rss: IntGauge,
+    start_time: IntGauge,
 }
 
 impl ProcessCollector {
     pub fn new(pid: libc::pid_t) -> Self {
         let mut descs = Vec::new();
 
-        let cpu_total = Counter::with_opts(Opts::new(
+        let cpu_total = IntCounter::with_opts(Opts::new(
```

Comment:
I'm afraid that this might break Grafana functionalities for existing users, if we change float counter to int counter.

Reply:
https://github.com/tikv/rust-prometheus/blob/d1528e11ab9dfe11a2528744ec7c42485b82ed1e/src/value.rs#L87-L102

```rust
match self.val_type {
    ValueType::Counter => {
        let mut counter = Counter::default();
        counter.set_value(val.into_f64());
        m.set_counter(counter);
    }
    ValueType::Gauge => {
        let mut gauge = Gauge::default();
        gauge.set_value(val.into_f64());
        m.set_gauge(gauge);
    }
}
```

the prometheus only has f64 numberic type link javascript, IntCounter/IntGauge finally would convert to f64

if we use IntGauge the benefit is only need to call once `as f64`

but if we use Gauge, all procfs int type must cast to f64 multitimes

Reply:
prometheus only has float/string two data type or literal from their document:

https://prometheus.io/docs/prometheus/latest/querying/basics/#float-literals